10_rest/nickolee/pkg/puppy/rest/rest.go

```diff
@@ -0,0 +1,130 @@
+package rest
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/anz-bank/go-course/10_rest/nickolee/pkg/puppy"
+
+	"github.com/go-chi/chi"
+	"github.com/go-chi/render"
+)
+
+const (
+	invalidInputMsg = "Invalid input, ensure ID and JSON are valid"
```

Comment:
This error message is used both when ID and JSON are invalid. Think about how you might separate these two error cases out to give the consumer a more accurate error message.

Reply:
fixed